cpp/arcticdb/storage/s3/s3_storage.cpp

```diff
@@ -80,19 +82,23 @@ bool S3Storage::do_key_exists(const VariantKey& key) {
 
 namespace arcticdb::storage::s3 {
 
-S3Storage::S3Storage(const LibraryPath &library_path, OpenMode mode, const Config &conf) :
+S3Storage::S3Storage(const LibraryPath &library_path, OpenMode mode, const S3Settings &conf) :
     Storage(library_path, mode),
     s3_api_(S3ApiInstance::instance()),  // make sure we have an initialized AWS SDK
     root_folder_(object_store_utils::get_root_folder(library_path)),
     bucket_name_(conf.bucket_name()),
     region_(conf.region()) {
-
     auto creds = get_aws_credentials(conf);
 
     if (conf.use_mock_storage_for_testing()){
         ARCTICDB_RUNTIME_DEBUG(log::storage(), "Using Mock S3 storage");
         s3_client_ = std::make_unique<MockS3Client>();
     }
+    else if (conf.aws_auth() == AWSAuthMethod::STS_PROFILE_CREDENTIALS_PROVIDER){
+        Aws::Config::ReloadCachedConfigFile(); // config files loaded in Aws::InitAPI; It runs once at first S3Storage object construct; reload to get latest
```

Comment:
Why is this reload here? I expected this to only be necessary within tests

Reply:
I think it's better to be reloaded everytime when `Arctic` is called for the sake of user-friendliness. Or user is needed to restarted the interpreter whenever there is a change of config (or called an API-pending to reload the config)